sema4ai/src/sema4ai_code/vendored_deps/package_deps/analyzer.py

```diff
@@ -441,6 +442,56 @@ def _iter_external_endpoints_issues(self) -> Iterator[_DiagnosticsTypedDict]:
                                     "message": f"Error: 'port' must be a valid integer between 0 and 65535 if provided in a rule. Found: {rule['port']!r} (type: {type_repr(rule['port'])})",
                                 }
 
+    def _iter_version_consistency_issues(self) -> Iterator[_DiagnosticsTypedDict]:
+        """Check if the package.yaml version matches the action package version in agent-spec.yaml."""
+        import yaml
+
+        data = self._yaml_data
+        if not data:
+            return
+
+        version = data.get("version")
+        if not version:
+            return  # Version check already reported in _load_yaml_info
+
+        agent_spec_path = (
+            pathlib.Path(self.path).parent.parent.parent.parent / "agent-spec.yaml"
+        )
+        if not agent_spec_path.exists():
+            return
+
+        try:
+            with agent_spec_path.open("r") as f:
+                agent_spec = yaml.safe_load(f)
+                action_packages = (
+                    agent_spec.get("agent-package", {})
+                    .get("agents", [{}])[0]
+                    .get("action-packages", [])
+                )
+
+                matching_action_inside_agent = None
+                for action in action_packages:
+                    if action.get("name") == data.get("name"):
+                        matching_action_inside_agent = action
+                        break
+
+                if matching_action_inside_agent:
```

Comment:
We could also notify that the action is not registered in the agent package at this point if `matching_action_inside_agent` is `None`...